### PR TITLE
[Bug] Fix transaction_fee in webhook_controller

### PR DESCRIPTION
**Pull request by Kenworthe**
- Should now properly exclude transaction_fee line items when iterating in webhooks controller
- Should fix email double-send.

### CODE REVIEW

File: app/controllers/webhooks_controller.rb [hunk 1]
```diff
@@ -114,10 +114,11 @@ def handle_payment_intent_succeeded(
     end
 
     items = JSON.parse(payment_intent.line_items)
-    recipient = payment_intent.recipient
```
Comment (Kenworthe) [inline app/controllers/webhooks_controller.rb:117]: unused var

File: app/controllers/webhooks_controller.rb [hunk 2]
```diff
@@ -114,10 +114,11 @@ def handle_payment_intent_succeeded(
...
     is_donation = false
     save_payment_intent = false
-    items = items.reject { |item| item['type'] == 'fee' }
+
+    items.reject! { |item| item['item_type'] == 'transaction_fee' }
```
Comment (justintmckibben) [inline app/controllers/webhooks_controller.rb:120]: You can merge this, but would be great to do a follow up that includes a test with a `transaction_fee` line item

**Review by justintmckibben — APPROVED** [review]

File: app/controllers/webhooks_controller.rb (hunk 2 above)
Comment (justintmckibben) [reply app/controllers/webhooks_controller.rb:120]: and in that test could you also add a test inside of the payment_intent_spec to test the amount on the intent?

Comment (Kenworthe) [reply app/controllers/webhooks_controller.rb:120]: yup. done! re-review?

Comment (justintmckibben) [reply app/controllers/webhooks_controller.rb:120]: Oh perf! yeah there is another test in the payment_intent_spec that tests the method "amount" to make sure that the amount of the payment intent is calculated properly, and I wanted to make sure that adding this new item_type doesn't break that

**Pull request merged**